### PR TITLE
Retargeting on VB tuple should retarget the underlying type

### PR DESCRIPTION
**Pull request by jcouv**
The tuple type should not go through normal retargeting logic, instead its underlying type should be retargeted and used.

@dotnet/roslyn-compiler for review.

Fixes crashing issue https://github.com/dotnet/roslyn/issues/13305
Relates to [this C# change ](https://github.com/dotnet/roslyn/pull/11047/files)

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/Symbols/Retargeting/RetargetingSymbolTranslator.vb [hunk 1]
```diff
@@ -149,7 +149,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols.Retargeting
             Private Function RetargetNamedTypeDefinition(type As NamedTypeSymbol, options As RetargetOptions) As NamedTypeSymbol
                 Debug.Assert(type Is type.OriginalDefinition)
 
-                ' Before we do anything else, check if we need to do special retargeting
+                If (type.IsTupleType) Then
```
Comment (cston) [inline src/Compilers/VisualBasic/Portable/Symbols/Retargeting/RetargetingSymbolTranslator.vb]: Parentheses are not necessary.

File: src/Compilers/VisualBasic/Portable/Symbols/Retargeting/RetargetingSymbolTranslator.vb [hunk 2]
```diff
@@ -149,7 +149,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols.Retargeting
...
+                    Return DirectCast(type, TupleTypeSymbol).WithUnderlyingType(Retarget(type.TupleUnderlyingType, options))
```
Comment (cston) [inline src/Compilers/VisualBasic/Portable/Symbols/Retargeting/RetargetingSymbolTranslator.vb]: The C# version has an extra check of `newUnderlyingType.IsTupleOrCompatibleWithTupleOfCardinality`. When does that fail and is the same check needed here?

Comment (jcouv) [inline src/Compilers/VisualBasic/Portable/Symbols/Retargeting/RetargetingSymbolTranslator.vb]: Thanks for pointing this out. I was looking at the PR that introduced the logic here first, but not the most recent version of the implementation.

There is one test that manages to produce a retargeted underlying type that isn't good: `NoPiaEmbedTypes.ErrorType_Tuple`. I'll add this check, but that test will only be ported later.

**Comment by cston** [comment]
LGTM

**Comment by gafter** [comment]
:shipit:

File: src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.vb [hunk 3]
```diff
@@ -210,8 +210,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 symbol = symbol.TupleUnderlyingType
             End If
 
-                If format.MiscellaneousOptions.IncludesOption(SymbolDisplayMiscellaneousOptions.UseErrorTypeSymbolName) AndAlso
-                String.IsNullOrEmpty(symbolName) Then
+            If format.MiscellaneousOptions.IncludesOption(SymbolDisplayMiscellaneousOptions.UseErrorTypeSymbolName) AndAlso
+            String.IsNullOrEmpty(symbolName) Then
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.vb]: I think this line should stay where it used to be.

Comment (jcouv) [inline src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.vb]: Right, yes.

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb [hunk 4]
```diff
@@ -4828,7 +4828,7 @@ End Namespace
 
         End Sub
 
-        <Fact(Skip:="https://github.com/dotnet/roslyn/issues/13305")>
+        <Fact>
         Public Sub MultipleDefinitionsOfValueTuple()
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb]: >MultipleDefinitionsOfValueTuple [](start = 19, length = 31)

I would expect to see unit-test(s) that specifically target multi-targeting. It looks like this one hits the scenario by accident and could easily be changed in a way that will stop that from happening.

Comment (jcouv) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb]: Added `RetargetTupleErrorType`

**Comment by AlekseyTs** [comment]
LGTM

**Comment by VSadov** [comment]
LGTM

**Pull request merged**